dts/bindings/gpio/microchip,mcp23xxx.yaml

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2020 Geanix ApS
+# Copyright (c) 2021 Peter Johanson
+# Copyright (c) 2021 metraTec GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+include: [gpio-controller.yaml]
+
+properties:
+  "#gpio-cells":
+    const: 2
+
+  int-gpios:
+    type: phandle-array
+    description: |
+        GPIO connected to the controller INT pin. This pin is active-low.
+
+  reset-gpios:
+    type: phandle-array
+    description: |
+        GPIO connected to the controller RESET pin. This pin is active-low.
+
+  ngpios:
```

Comment:
Hmmm.... This looks a bit suspicious to me.

From what I understand, `ngpios` is supposed to be selectable on a per-system basis and should not be a `const` or `enum`.

E.g. a particular board or system is implemented using only the first 3 gpio even though the controller supports 32. So the controller's mask of available gpio would be `~7`.

The DTS binding description seems to agree with the above interpretation.

I would suggest using a separate device tree property to indicate how many gpio are supported by the controller.

Reply:
Note that I didn't actually create this property, I simply merged `microchip,mcp23sxx.yaml` and `microchip,mcp230xx.yaml` into one file. The existing driver internally expects this property to be either `8` or `16`, changing this would be quite a bit of effort.

Reply:
@cfriedt I believe that is not the correct interpretation of the `ngpios` property. From the `gpio-controller.yaml` binding:

        This property indicates the number of in-use slots of available slots
        for GPIOs. The typical example is something like this: the hardware
        register is 32 bits wide, but only 18 of the bits have a physical
        counterpart. The driver is generally written so that all 32 bits can
        be used, but the IP block is reused in a lot of designs, some using
        all 32 bits, some using 18 and some using 12. In this case, setting
        "ngpios = <18>;" informs the driver that only the first 18 GPIOs, at
        local offset 0 .. 17, are in use.  For cases in which there might be
        holes in the slot range, this value should be the max slot number-1.

So it is not intended to be selectable on a per-system basis, but rather on a per-instance basis where some instances support a different _physical_ number of pins.